### PR TITLE
build: add coverage and audit to github actions

### LINKED ISSUE #1513: ci tooling change?
See:

- https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
- https://blog.travis-ci.com/oss-announcement

Travis is looking like slightly shaky ground now.

Is it worth giving GitHub Actions a try?

cc @charlierudolph @aslakhellesoy

### PR DESCRIPTION
**Pull request by davidjgoss**
Closes #1513

Coverage is being reported to coveralls successfully, I think someone with admin privileges needs to change a setting there for it to start commenting on PRs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -229,7 +229,6 @@
     "chai-exclude": "2.0.2",
     "coffeescript": "2.5.1",
     "core-js": "3.8.1",
-    "coveralls": "3.1.0",
```
Comment (davidjgoss) [inline package.json:232]: We don't need this as a dependency now because the coveralls action handles it.

File: package.json [hunk 2]
```diff
@@ -270,7 +269,8 @@
     "precck-test": "yarn run build-local",
     "prefeature-test": "yarn run build-local",
     "prepublishOnly": "rm -rf lib && npm run build-local",
-    "test-coverage": "yarn run lint && ./scripts/test-coverage.sh",
+    "pretest-coverage": "yarn build-local",
+    "test-coverage": "nyc --silent mocha 'src/**/*_spec.ts' 'compatibility/**/*_spec.ts' && nyc --silent --no-clean node ./bin/cucumber-js && nyc report --reporter=lcov",
```
Comment (davidjgoss) [inline package.json:273]: This seemed to fit reasonably as a package script now, but happy to break it down or flip back to shell if preferred.

**Review by charlierudolph — APPROVED** [review]
Awesome! Looks great

**Comment by davidjgoss** [comment]
Thanks! Think we just need to deactivate appveyor now.

Looks like the slight decrease in coverage reported by coveralls is in `src/formatter/get_color_fns.ts` so maybe a difference in shells?

**Pull request merged**